backend/app/requests/representative_points.py

```diff
@@ -58,7 +58,10 @@ def representative_points_request(app, flask_request, engine):
         )
     except (json.JSONDecodeError, KeyError):
         raise InvalidFormat(message='Invalid JSON format.')
+
+    dataset_hint = request_json.get('dataset_hint', None)
```

Comment:
If I understand this correctly, this means that having two different datasets for the same state will not share a cache file for RPs.